lib/win32/certstore/certstore.rb

```diff
@@ -15,15 +15,17 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-require "win32/certstore/version"
-require 'win32/api/reserved_names'
+require 'win32/store/assertions'
+require 'win32/store/crypto'
 
 module Win32
-  module Certstore
-    include Win32::API::ReservedNames
-    include Chef::Mixin::WideString
+  class Certstore
+    extend Win32::Store::Assertions
+    extend Win32::Store::Crypto
+    extend Chef::Mixin::ShellOut
+    extend Chef::Mixin::WideString
 
-    def open store_name
+    def self.open_store(store_name)

```

Comment:
Most of the changes look good to me. Just 2 doubts:
1. All the methods defined inside `Certstore` are class methods. Does this bind us to define only class methods in `Certstore` in future?
2. After renaming the method user will call `Win32::Certificate.open_store(certstore_name)`. It would look better to call  `Win32::Certificate.open(certstore_name)`. Here if it's a Class method, it won't override Ruby's `open` method since this will always be called along with the Class name.

@btm, @mwrock please suggest if this looks fine or some modifications should be done.

Reply:
I think a class method of `open` could instantiate an instance of store and return it. One would call `list` and other instance methods of of that. `open` could also take a block so you could have:

```
Win32::Certstore.open("MY") do |store| 
    store.list
end
```
If called with a block, you could call `close` on the store. It might also be worth closing the store in a finalizer.